lib/ansible/modules/windows/win_wait_for_process.ps1

```diff
@@ -64,99 +64,108 @@ Function Get-ProcessMatchesFilter {
         [int]
         $ProcessId
     )
-    
-    $CIMProcesses = Get-CimInstance Win32_Process
-    foreach ($CIMProcess in $CIMProcesses)
-    {
-        $include = $true
-        if(-not [String]::IsNullOrEmpty($ProcessNamePattern))
-        {
-            #if a process name was specified in the filter, validate that here.
-            $include = $include -and ($CIMProcess.ProcessName -match $ProcessNamePattern)
-        }
-        if($ProcessNameExact -is [Array] -or (-not [String]::IsNullOrEmpty($ProcessNameExact)))
-        {
-            #if a process name was specified in the filter, validate that here.
-            if ($ProcessNameExact -is [Array] )
-            {
-                $include = $include -and ($ProcessNameExact -contains $CIMProcess.ProcessName)
-            }   
-            else {
-                $include = $include -and ($ProcessNameExact -eq $CIMProcess.ProcessName)
+
+    $WMIProcesses = Get-Process -IncludeUserName
+    foreach ($WMIProcess in $WMIProcesses) {
+
+        # If a process name was specified in the filter, validate that here.
+        if (-not [String]::IsNullOrEmpty($ProcessNamePattern)) {
```

Comment:
You don't have to change it but this can be simplified with `if ($ProcessNamePattern) {`.

Reply:
Yup, I prefer to change that ;-)